pkg/oomstore/revision.go

```diff
@@ -24,6 +25,46 @@ func (s *OomStore) GetRevisionBy(ctx context.Context, groupID int, revision int6
 	return s.metadata.GetRevisionBy(ctx, groupID, revision)
 }
 
+func (s *OomStore) CreateRevision(ctx context.Context, opt metadata.CreateRevisionOpt) (int, string, error) {
+	var revisionID int
+	var snapshotTable string
+	var dummyRevision *types.Revision
+
+	if err := s.metadata.WithTransaction(ctx, func(c context.Context, db metadata.DBStore) error {
+		_, err := s.metadata.GetRevisionBy(ctx, opt.GroupID, 0)
+		if err != nil {
+			if !errdefs.IsNotFound(err) {
+				return err
+			}
+
+			if _, _, err = s.metadata.CreateRevision(ctx, metadata.CreateRevisionOpt{
+				Revision:    0,
+				GroupID:     opt.GroupID,
+				Description: "dummy revision will be used at Join and Export",
+			}); err != nil {
+				return err
+			}
+
+			if dummyRevision, err = s.metadata.GetRevisionBy(ctx, opt.GroupID, 0); err != nil {
+				return err
+			}
+		}
+
+		revisionID, snapshotTable, err = s.metadata.CreateRevision(ctx, opt)
+		return err
+	}); err != nil {
+		return 0, "", err
+	}
+
+	if dummyRevision != nil {
```

Comment:
can you move the logic right after line 51?

Reply:
also, `createFirstSnapshotTable` needs to be deprecated; for dummy revision, we need to create both snapshot table and cdc table

Reply:
- can you move the logic right after line 51?

Moving after line 51 causes the offline store SQL to be executed in the meta store transaction, which would be a problem in SQLite, so it's written here to avoid that, similar to apply's onlineJobs.

Reply:
ah I see!

Reply:
- also, createFirstSnapshotTable needs to be deprecated; for dummy revision, we need to create both snapshot table and cdc table

I created a new function: `createSnapshotAndCdcTable` instead of `createFirstSnapshotTable`